### PR TITLE
ibus: fix dconf database

### PR DESCRIPTION
**Pull request by symphorien**
###### Motivation for this change

#85892 was reverted because of an evaluation error which manifested among other in xfce test.
This is a fixed PR where the xfce test evaluates.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s): ibus and xfce
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by symphorien** [comment]
cc @worldofpeace @jtojnar

**Comment by jtojnar** [comment]
Is https://github.com/NixOS/nixpkgs/commit/bd6c12ba32b1b076baee6cfc0f1fc80b22e04987#diff-a149871f918ec698c20b8b19f3f2e6d6 no longer necessary?

**Comment by jtojnar** [comment]
Otherwise it still look reasonable to me at a glance but did not test it.

**Comment by symphorien** [comment]
Oh, indeed I lost some commits with git. I will fix this.

**Comment by symphorien** [comment]
This passes the ibus and xfce test, and I checked in a vm that ibus-daemon does not warn about dconf anymore.

File: nixos/modules/programs/dconf.nix [hunk 1]
```diff
@@ -4,13 +4,24 @@ with lib;
...
+      mkdir -p $out/profile
+      mkdir -p $out/db
+    '' + (
+      concatStringsSep "\n" (
+        mapAttrsToList (
+          name: path: ''
+            ln -s ${path} $out/profile/${name}
+          ''
+        ) cfg.profiles
+      )
```
Comment (ttuegel) [inline nixos/modules/programs/dconf.nix:20]: Why won't the profiles already be linked by `symlinkJoin`?

Comment (jtojnar) [reply nixos/modules/programs/dconf.nix:20]: I was also confused by this but profiles are not packages, they are just file paths.

Comment (symphorien) [reply nixos/modules/programs/dconf.nix:20]: yes, the symlinkJoin is only for the option `packages` and this look is for the option `profiles`. So I think this is correct.

Comment (ttuegel) [reply nixos/modules/programs/dconf.nix:20]: I understand, thanks!

**Review by ttuegel — APPROVED** [review]

**Pull request merged**